File: packages/agent/src/infrastructure/events/types/file.ts
/**
 * @fileoverview File Events
 *
 * Events for file read, write, and edit operations.
 */

import type { BaseEvent } from './base.js';

// =============================================================================
// File Events
// =============================================================================

/**
 * File read event
 */
export interface FileReadEvent extends BaseEvent {
  type: 'file.read';
  payload: {
    path: string;
    lines?: { start: number; end: number };
  };
}

/**
 * File write event
 */
export interface FileWriteEvent extends BaseEvent {
  type: 'file.write';
  payload: {
    path: string;
    size: number;
    /** Content hash for deduplication */
    contentHash: string;
  };
}

/**
 * File edit event
 */
export interface FileEditEvent extends BaseEvent {
  type: 'file.edit';
  payload: {
    path: string;
    oldString: string;
    newString: string;
    /** Patch/diff representation */
    diff?: string;
  };
}

File: packages/agent/crates/tron-events/src/types/payloads/file.rs
//! File event payloads: read, write, edit.

use serde::{Deserialize, Serialize};

/// Payload for `file.read` events.
#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct FileReadPayload {
    /// File path.
    pub path: String,
    /// Line range read.
    #[serde(skip_serializing_if = "Option::is_none")]
    pub lines: Option<LineRange>,
}

/// Line range for file read.
#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
pub struct LineRange {
    /// Start line (1-based).
    pub start: i64,
    /// End line (inclusive).
    pub end: i64,
}

/// Payload for `file.write` events.
#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct FileWritePayload {
    /// File path.
    pub path: String,
    /// File size in bytes.
    pub size: i64,
    /// Content hash.
    pub content_hash: String,
}

/// Payload for `file.edit` events.
#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct FileEditPayload {
    /// File path.
    pub path: String,
    /// Old string replaced.
    pub old_string: String,
    /// New string inserted.
    pub new_string: String,
    /// Diff output.
    #[serde(skip_serializing_if = "Option::is_none")]
    pub diff: Option<String>,
}
